src/tests/index.test.ts

```diff
@@ -247,6 +247,14 @@ describe('matchJsonToSchema.test.ts', () => {
       },
       true,
     ],
+    [{ test: 'else' }, { test: { $exist: true } }, true],
+    [{ test: 'else' }, { test: { $exist: false } }, false],
+    [{ test1: 'else' }, { test: { $exist: true } }, false],
+    [{ test1: 'else' }, { test: { $exist: false } }, true],
+    ['/test', '/test', true],
+    ['/test', '/test2', false],
+    [1, 1, true],
+    [1, 2, false],
```

Comment:
I think we should have a more complex case of mixing $or / $and with $not  and $exist deeply nested for sanity